src/api/syscall.c

```diff
@@ -188,20 +188,8 @@ exception_t handleUnknownSyscall(word_t w)
     }
 #endif
 
-#ifdef CONFIG_ENABLE_BENCHMARKS
-    if (w == SysBenchmarkFlushCaches) {
-#ifdef CONFIG_ARCH_ARM
-        tcb_t *thread = NODE_STATE(ksCurThread);
-        if (getRegister(thread, capRegister)) {
-            arch_clean_invalidate_L1_caches(getRegister(thread, msgInfoRegister));
-        } else {
-            arch_clean_invalidate_caches();
-        }
-#else
-        arch_clean_invalidate_caches();
-#endif
-        return EXCEPTION_NONE;
-    } else if (w == SysBenchmarkResetLog) {
+#if defined(CONFIG_ENABLE_BENCHMARKS) || defined(CONFIG_KERNEL_LOG_BUFFER)
```

Comment:
This looks incorrect to me. In config.cmake above, the constant is `KERNEL_DEBUG_LOG_BUFFER` and here it's `KERNEL_LOG_BUFFER`. IMO the latter is better if you have to choose, it's not just for debugging.

Reply:
Or are these two different buffers? I'm confused

Reply:
It should be the same buffer, I think you're correct and that's using the wrong define, also that `KERNEL_LOG_BUFFER` makes more sense given it's the same buffer either way.

Reply:
The only difference should be the message format in the buffer and the external structure used to track & iterate through messages inside and outside the kernel.

Reply:
* [ ] `KERNEL_LOG_BUFFER` for configuration definition

Reply:
We should call the new feature `CONFIG_KERNEL_EVENT_TRACING` to avoid naming confusion.

Following the logic of other automatically generated config settings (`CONFIG_ENABLE_BENCHMARKS`, `CONFIG_ENABLE_SMP_SUPPORT`), the name `CONFIG_KERNEL_LOG_BUFFER` could become `CONFIG_ENABLE_KERNEL_LOG_BUFFER`. However, that would change a lot of places, so I'm not sure how much backing this gets.